packages/grafana-data/src/datetime/moment_wrapper.ts

```diff
@@ -123,7 +123,12 @@ export const dateTimeForTimeZone = (
 };
 
 export const getWeekdayIndex = (day: string) => {
-  return moment.weekdays().findIndex((wd) => wd.toLowerCase() === day.toLowerCase());
+  const prevLocale = getLocale();
+  setLocale('en');
+  const index = moment.weekdays().findIndex((wd) => wd.toLowerCase() === day.toLowerCase());
+  setLocale(prevLocale);
+
+  return index;
```

Comment:
**Suggestion:** although I am usually against using comments, I think here we could leave some explanation on the _why_ (as that's probably going to get "lost" later).

Alternatively - as the server returns these in English (currently) - we could also just do a much dumber, although simpler option:
```javascript
export const getWeekdayIndexByEnglishName(day: string) => [
    'Sunday', 
    'Monday', 
    'Tuesday', 
    'Wednesday', 
    'Thursday', 
    'Friday', 
    'Saturday'
].findIndex(wd => wd.toLowerCase() === day.toLowerCase());
```

Reply:
Great suggestion, thanks! It's easier to understand and there are less unintended side-effect risks. I made the update.